File: BOJ 2/10866.cpp
#include <cstdio>
#include <deque>
#include <string.h>
using namespace std;
deque<int> dq;
int n;
int main(){
	scanf("%d", &n);
	while (n--) {
		char s[20] = { '\0', };
		scanf("%s", s);
		if (!strcmp(s, "push_back")) {
			int d;
			scanf("%d", &d);
			dq.push_back(d);
		}
		else if (!strcmp(s, "push_front")) {
			int d;
			scanf("%d", &d);
			dq.push_front(d);
		}
		else if (!strcmp(s, "front")) {
			if (dq.empty()) printf("-1\n");
			else printf("%d\n", dq.front());
		}
		else if (!strcmp(s, "back")) {
			if (dq.empty()) printf("-1\n");
			else printf("%d\n", dq.back());
		}
		else if (!strcmp(s, "size")){
			if(!dq.empty()) printf("%d\n", dq.size());
			else printf("0\n");
		}
		else if (!strcmp(s, "empty")) {
			if (dq.empty()) printf("1\n");
			else printf("0\n");
		}
		else if (!strcmp(s, "pop_front")) {
			if (dq.empty()) printf("-1\n");
			else {
				printf("%d\n", dq.front());
				dq.pop_front();
			}
		}
		else if (!strcmp(s, "pop_back")) {
			if (dq.empty()) printf("-1\n");
			else {
				printf("%d\n", dq.back());
				dq.pop_back();
			}
		}
	}
	return 0;
}

File: BOJ(Python 공부용)/10866.py
#!/usr/bin/env python

import sys
import collections
deque = collections.deque()
n = input()
n = int(n)
for i in range(n):
    choice = sys.stdin.readline().strip()
    if choice[:10] == "push_front":
        num = int(choice[11:])
        deque.appendleft(num)
    if choice[:9] == "push_back" :
        num = int(choice[10:])
        deque.append(num)
    if choice[:9] == "pop_front" :
        if not deque:
            print("-1")
            continue
        print("%d" %deque.popleft())
    if choice[:8] == "pop_back":
        if not deque:
            print("-1")
            continue
        print("%d" %deque.pop())
    if choice[:4] == "size":
        print("%d" %len(deque))
    if choice[:5] == "empty":
        if deque:
            print("0")
            continue
        print("1")
    if choice[:5] == "front":
        if not deque:
            print("-1")
            continue
        print("%d" %deque[0])
    if choice[:4] == "back":
        if not deque:
            print("-1")
            continue
        print("%d" %deque[-1])